CMakeModules/InternalUtils.cmake

```diff
@@ -68,6 +68,7 @@ macro(arrayfire_set_cmake_default_variables)
 
   set(CMAKE_CXX_STANDARD 11)
   set(CMAKE_CXX_EXTENSIONS OFF)
+  set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```

Comment:
This is our default visibility unless we explicitly add visibility attribute for the declarations we want to expose. Is that correct ?